types/index.d.ts

```diff
@@ -58,6 +60,26 @@ export class Asset {
   issuer: string;
 }
 
+export class ChangeTrustAsset {
+  static native(): ChangeTrustAsset;
+  static creditAsset(code: string, issuer?: string): ChangeTrustAsset;
+  static liquidityPoolAsset(liquidityPoolParameters: LiquidityPoolParameters): ChangeTrustAsset;
+  static fromOperation(xdr: xdr.ChangeTrustAsset): ChangeTrustAsset;
+
+  getCode(): string;
+  getIssuer(): string;
+  getLiquidityPoolParameters(): LiquidityPoolParameters;
+  getAssetType(): AssetType;
+  isNative(): boolean;
+  isLiquidityPool(): boolean;
+  equals(other: ChangeTrustAsset): boolean;
+  toXDRObject(): xdr.ChangeTrustAsset;
+
+  code: string;
+  issuer: string;
+  liquidityPoolParameters: LiquidityPoolParameters;
+}
+
```

Comment:
Again, maybe this should inherit from `Asset`, or otherwise have some common interface?

Reply:
I'm trying inheritance in this commit: https://github.com/stellar/js-stellar-base/pull/454/commits/83ab66bc0ba406607c06b4328f98a51ad14bc448. I'm not very happy with it though because the `Asset` class has become a bit dirty as some method signatures were updated to accept not only Asset but also ChangeTrustAsset.

The other options could include generics (I think it'd get even dirtier) and some level of feature composition... these are not topics I'm experienced in JS or TS though so I wanted to hear from you before diving into that.

Reply:
This is a tough call.. on one hand there is much less duplicated code in this commit (nice!) but as you say it does require a little bit of shoe-horning in terms of the type signatures. I do think this version will be easier to maintain (since all validation is now consolidated), and I left a few more thoughts in the commit itself.

Reply:
We agreed (offline) to handle the code repetition in a separate PR to make it easier to review and discuss.